### ISSUE TITLE
Add workaround for `QueryProcessor` race condition [CASSANDRA-17401]

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/stargate#1655]
Looks like there is a new rare (but occurring) race condition for `QueryProcessor`; for details see:

https://issues.apache.org/jira/browse/CASSANDRA-17401

this affects latest Cassandra versions at this point (3.11.12, 4.0.3) and since we may not be able to wait for the fix, let's try to work around the problem by using a simple retry.

**Comment by tatu-at-datastax** [comment]
Attempted to reproduce the issue as suggested in #1647 but I don't think this actually reproduces the issue we are trying to fix, because:

1. I do not see anything indicating the underlying failure in IT logs -- only incorrect number of messages (1000 instead of 2000)
2. Adding suggested fix does not change behavior in any way.

@ivansenic @mpenick will sync up with you tomorrow to see if you have ideas on how to proceed here.

I also noticed that the fix for 3.11 needs to be slightly different from 4.0, due to differences in `QueryProcess.getStatement()` between 3.11/4.0 implementations. But I assume that is doable so I first focused on 4.0.
